cmd/otelcol/config/collector/agent_config.yaml

```diff
@@ -123,7 +123,11 @@ processors:
       #- action: insert
         #value: staging/production/...
         #key: deployment.environment
-
+  resource/add_mode:
+    attributes:
+      - action: insert
+        value: "agent"
+        key: service_mode
```

Comment:
I know there is no existing conventions for this, but OTel spec recommends the resource attribute names to be specific enough to figure out what entity they are part of, using namespacing. I think something like `otelcol.service.mode` whould be better choice here.

Reply:
PR updated to relfect the semantic naming conventions, thank you for your insight ... I originally went off `service_version`.